src/components/structures/RoomView.js

```diff
@@ -1052,13 +1061,19 @@ module.exports = React.createClass({
                     );                
                 }
                 else {
+                    var inviterName = undefined;
+                    if (this.props.oobData) {
+                        inviterName = this.props.oobData.inviterName;
+                    }
+
                     return (
                         <div className="mx_RoomView">
-                            <RoomHeader ref="header" room={this.state.room} simpleHeader="Join room"/>
```

Comment:
I made RoomHeader drop back to displaying the simple, 'Join Room' message if it really can't get any information about the room, which should make it redundant here (I'd be in favour of splitting out the simple header stuff into a separate component that's used by RoomHeader and then directly in the other places that use RoomHeader's simpleHeader).